lib/mix/lib/mix/tasks/format.ex

```diff
@@ -360,10 +360,20 @@ defmodule Mix.Tasks.Format do
     map =
       for input <- List.wrap(formatter_opts[:inputs]),
           file <- Path.wildcard(Path.join(prefix ++ [input]), match_dot: true),
-          do: {file, formatter_opts},
+          do: {file, {dot_formatter, formatter_opts}},
           into: %{}
 
-    Enum.reduce(subs, Map.merge(acc, map), fn {sub, formatter_opts_and_subs}, acc ->
+    acc =
+      Map.merge(acc, map, fn file, {dot_formatter1, _}, {dot_formatter2, formatter_opts} ->
+        Mix.shell().error(
+          "Both #{dot_formatter1} and #{dot_formatter2} specify the file #{file} " <>
+            "in their :inputs option"
```

Comment:
Should we clearly say in the error message that options from `dot_formatter2` are used when there's a conflict like this?

Reply:
Extra info won’t hurt 👍

Reply:
On other hand, that message might get outdated if there are other conflicting `.formatter.exs` files. It would be quite an edge case, but possible.

Here is an example:

```
.formatter.exs # [inputs: "lib/**/*.ex", subdirectories: ["lib"]]
lib/.formatter.exs # [inputs: ["*.ex", "foo/**/*.ex"], subdirectories: ["foo"]]
lib/foo/.formatter.exs # [inputs: "*.ex"]
lib/foo/a.ex
```

Reply:
The last one of all of those will win for a given file, no?